modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -423,14 +436,32 @@ def _compute_axis_labels(self, axis: int, partitions=None):
         -------
         pandas.Index
             Labels for the specified `axis`.
+        List of int
+            Size of partitions alongsize specified `axis`.
         """
         if partitions is None:
             partitions = self._partitions
-        new_index, _ = self._partition_mgr_cls.get_indices(axis, partitions)
-        return new_index
+        new_index, internal_idx = self._partition_mgr_cls.get_indices(axis, partitions)
+        return new_index, list(map(len, internal_idx))
```

Comment:
```suggestion
        return new_index, internal_idx
```
Should we return external and internal labels since the function name says exactly about that? We will have many `list(map(len, internal_idx))` in different places though.

Reply:
This is the reason why I decided to implement this `list(map())` here - to avoid adding it everywhere else.
Maybe you have a better naming idea for the function?

Reply:
We already have similar things in `get_indices` (AFAIRC introduced by you too 😄). We should probably do there the same as here.

As to the function name, maybe just `_compute_axis`?

Reply:
`get_indices()` is slightly different, living in a different layer and returning indices (as its name implies).
So I don't like the vagueness of `_compute_axis`, though, so what about `_compute_axis_labels_and_size` or `_compute_axis_labels_and_splits`?

Reply:
`_compute_axis_labels_and_splits` is fine to me. Another example would be `_compute_axis_labels_and_lengths`? as we use `lengths` word in many places. Do you think which is better?

Reply:
renamed to `_compute_axis_labels_and_lengths()`, this expresses the returns better (though it added a few lines due to some lines getting long and `black` reformatting)